### ISSUE TITLE
Define sequential navigation for imagemaps

### ISSUE THREAD

**Issue by LJWatson** [w3c/html#297]
Migrated from <a href="https://www.w3.org/Bugs/Public/show_bug.cgi?id=27787">Bugzilla #27787</a>.

Ping @rodneyrehm @chaals

**Comment by rodneyrehm** [comment]
## Original issue (2015-01-08)

"User agents may also allow individual area elements representing hyperlinks to be selected and activated (e.g. using a keyboard)."

Every major user agent (Gecko, Trident, WebKit, Blink) adds `<area>` elements to the document's sequential navigation focus order. But because the specification does not explain _how_ that should happen, they disagree on the order.

In Blink and WebKit the `<area>` elements are added to the sequence in DOM order. In Gecko and Trident `<area>` themselves are inserted into the sequence in place of the `<img usemap="">` actually using the map containing the `<area>` elements.

So in Blink and WebKit the image maps are accessible, just not in place of the image, but wherever they appear in the document. The `<map>` element might be placed right after `<body>`, but the `<img usemap>` may be used in a modal `<dialog>` element. There is no way to Tab through the image's map. (using the `<dialog>` here is just an extreme example to make a point.)

"Because a map element (and its area elements) can be associated with multiple img and object elements, it is possible for an area element to correspond to multiple focusable areas of the document."

In Trident those multiple references to the same map cause the the associated `<area>` elements to occur multiple times in the document's sequential navigation focus order. In Gecko this should happen, but is masked by a bug. In Blink and WebKit the `<area>` elements are only tabbed through once.

Can we get the Trident behavior added to the specification?

## Reply from @chaals (2015-01-09)

Will run this past the accessibility task force, but it seems to make sense that the relationship between a label and a form control is roughly the same as an image and an image map, with the proviso that area elements are not rendered by default - so as if a label applied to a form button that was not, itself, going to be rendered…

**Comment by chaals** [comment]
Hmm. Thinking, it seems to make sense to specify the trident/gecko approach if we decide to put this into the material about focus navigation.

It's analogous to the way `label` elements "should" behave, **IMHO**, although currently a `label` cannot refer to more than one control, since you can only activate the first wrapped control, and if you use a `for` attribute it's an error to have a repeated `name` or `id`.

**Comment by dylanb** [comment]
Trident approach makes much more sense from a usability/accessibility perspective, so this has my +1

**Comment by chaals** [comment]
The number one question IMHO is whether the difference causes a real problem.

Question zero being how common this case is in the first place…

I've started [a public-html@ thread](http://www.w3.org/mid/op.yhyuk8zes7agh9@widsith.local), [a WAI Interest Group one](http://www.w3.org/mid/op.yhyuk9ips7agh9@widsith.local), and one on the WebAIM discussion list. Let's see if we get feedback saying we need to address this. My suspicion is that it is low priority for people. If that's the case, we're unlikely to get implementation in a hurry :(

**Comment by chaals** [comment]
The trident approach makes more sense, but user agents *do* allow keyboard access already, and this seems to be a vanishingly unimportant legacy concern. I will propose a minor tweak, to say that user agents *should* enable keyboard activation instead of the current *may*, and close, unless there is an objection.

The reason for not making it *must* is that while no user agent I know of properly handles the case of multiple redundant links, that's potentially a reasonable behaviour, so I don't see much value in making it explicitly non-conformant.
